posthog/models/filters/mixins/simplify.py

```diff
@@ -25,21 +26,34 @@ def simplify(self: T, team: "Team", **kwargs) -> T:
 
         # :TRICKY: Make a copy to avoid caching issues
         result: Any = self.with_data({"is_simplified": True})  # type: ignore
+
         if getattr(result, "filter_test_accounts", False):
-            result = result.with_data(
-                {"properties": result.properties + team.test_account_filters, "filter_test_accounts": False,}
+
+            new_group = {"type": "AND", "values": team.test_account_filters}
+            prop_group = (
+                {"type": "AND", "values": [new_group, result.property_groups.to_dict()]}
+                if result.property_groups.to_dict()
+                else new_group
             )
+            result = result.with_data({"properties": prop_group, "filter_test_accounts": False,})
 
         updated_entities = {}
         if hasattr(result, "entities_to_dict"):
             for entity_type, entities in result.entities_to_dict().items():
                 updated_entities[entity_type] = [self._simplify_entity(team, entity_type, entity, **kwargs) for entity in entities]  # type: ignore
 
-        properties = self._simplify_properties(team, result.properties, **kwargs)  # type: ignore
+        prop_group = self._simplify_property_group(team, result.property_groups, **kwargs).to_dict()  # type: ignore
```

Comment:
Q: why are we doing the property group manipulations using the dict representation?